### PR TITLE
PM12 IN-1172 Pending report type assignments

### PR DESCRIPTION
**Pull request by townxelliot**
## Purpose

[IN-1172](https://opgtransform.atlassian.net/browse/IN-1172)

## Approach

* Apply after the casesupervisionlevel fixes (earlier PM numbers).
* Remove existing report type assignments for annual_report_logs.
* Use casesupervisionlevel (set by other PM fixes) to add a report type assignment for each PENDING report that we migrated, provided it doesn't already have a report type assignment.

## Learning

n/a

## Checklist

* [ ] I have performed a self-review of my own code
* [ ] I have done an adhoc run against preprod (only needed for high complexity PRs)
* [ ] I have added relevant logging with appropriate levels to my code
* [ ] I have updated documentation where relevant
* [ ] I have added tests to prove my work
* [ ] The product team have tested these changes

### CODE REVIEW

File: scripts/post_migration_fixes/pmf_report_type_assignments_20220308_in1172/02_annual_report_type_assignments_insert.sql [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    SELECT
+        nextval('annual_report_type_assignments_id_seq') AS id,
+        annualreport_id,
+        (CASE
+            WHEN supervisionlevel = 'GENERAL' THEN 'OPG102'
+            WHEN supervisionlevel = 'MINIMAL' THEN 'OPG103'
+            ELSE NULL
+        END) AS reporttype,
+        'pfa' AS type
```
Comment (townxelliot) [inline scripts/post_migration_fixes/pmf_report_type_assignments_20220308_in1172/02_annual_report_type_assignments_insert.sql:16]: Use Sirius cases.casesubtype to set type.

Comment (townxelliot) [inline scripts/post_migration_fixes/pmf_report_type_assignments_20220308_in1172/02_annual_report_type_assignments_insert.sql:16]: Use Sirius cases.casesubtype to set type.

**Review by jackgoodby — DISMISSED** [review]

**Review by jackgoodby — DISMISSED** [review]

**Review by jackgoodby — APPROVED** [review]
Unclear where casrec_mapping.annual_report_logs are coming from,
may be worth bringing into this SQL?

**Review by jackgoodby — DISMISSED** [review]

**Comment by townxelliot** [comment]
This has been applied to preqa.

**Comment by townxelliot** [comment]
> Unclear where casrec_mapping.annual_report_logs are coming from, may be worth bringing into this SQL?

It's not possible to do that here, as they have to be copied out of the integration db then copied into the casrec_mapping schema. post_migration_scripts.py has the equivalent code for automating this step locally.

**Review by jackgoodby — DISMISSED** [review]

**Review by jamesrwarren — APPROVED** [review]

**Review by jamesrwarren — APPROVED** [review]

**Comment by townxelliot** [comment]
This was applied to preqa 2022-03-24 15:10.

**Review by jackgoodby — APPROVED** [review]

**Pull request merged**